sycl/include/sycl/detail/vector_arith.hpp

```diff
@@ -477,6 +500,172 @@ template <typename Self> struct VecOperators {
                     OpAssign<ShiftRight>, IncDec> {};
 };
 
+#if !__SYCL_USE_LIBSYCL8_VEC_IMPL
+template <typename Self> struct SwizzleOperators {
```

Comment:
Nit; The `Self` name is used all around to represent a shared-pointer to the implementation itself. Could we maybe call this `SelfT` to disambiguate?

Reply:
I think it's a bit late: https://github.com/search?q=repo%3Aintel%2Fllvm+Self+path%3Asycl%2Finclude%2Fsycl%2F&type=code. I also don't see an issue with that. IMO, `Self` is like `this` but just a bit different. shared-pointer impl or CRTP both satisfy that view.

Reply:
> IMO, Self is like this but just a bit different. shared-pointer impl or CRTP both satisfy that view.

That's my point. The shared-pointer impl is like `this` but different, and I don't see a reflective type being as close. Even if you could make the argument that it is, it creates the ambiguity that I wanted to avoid. Alas, it's just a nit, so I won't fight. 😉